reactor-core/src/main/java/reactor/core/publisher/MonoMetrics.java

```diff
@@ -132,6 +131,7 @@ final public void onNext(T t) {
 				return;
 			}
 			done = true;
+			//TODO looks like we don't count onNext: `Mono.empty()` vs `Mono.just("foo")`
 			FluxMetrics.recordOnComplete(commonTags, registry, subscribeToTerminateSample);
```

Comment:
@smaldini shouldn't we increment an `onNext` counter here, in order to differentiate empty case vs valued case? (this is not a regression and potentially a candidate for an `enhancement` follow-up issue)

Reply:
(removed the comment and moved that in a separate issue #1803)